lib/pbench/server/database/models/server_config.py

```diff
@@ -112,6 +112,23 @@ def validate_lifetime(key: str, value: JSONVALUE) -> JSONVALUE:
     return days
 
 
+# Define the minimum and maximum allowed length for a dataset name.
+DATASET_NAME_LEN_MIN = "min"
+DATASET_NAME_LEN_MAX = "max"
+
+
+def validate_name_len(key: str, value: JSONVALUE) -> JSONVALUE:
+    try:
+        min = int(value[DATASET_NAME_LEN_MIN])
+        max = int(value[DATASET_NAME_LEN_MAX])
+    except (KeyError, TypeError, ValueError):
+        raise ServerConfigBadValue(key, value)
+    else:
+        if min <= 0 or max < min or max > 1024:
+            raise ServerConfigBadValue(key, value)
+    return value
```

Comment:
Ugh; much harder to read and I see no benefit. (Even aside from the fact that your `ValueError` message would be lost.)

The use of `try/else` is an interesting quirk: it's mostly useful when there's a dependent follow-on that you don't want inside the same `try/except` scope, which isn't really true here. I used the idiom, but putting the test outside the `try` would have worked fine. As you perhaps inadvertently demonstrated, putting it *inside* requires a bit more finesse, though it could have just raised `ServerConfigBadValue` ... and the fact that the current exception interface isn't designed to provide detailed diagnosis is a separate issue entirely.

Reply:
> much harder to read and I see no benefit.

Fair enough.  To my eye, when the relational operators all line up like that, it's actually easier to read/comprehend than the corresponding list of `and` or `or` conditions, but nevermind.

> aside from the fact that your `ValueError` message would be lost.

Actually, I believe that it will be chained to the `ServerConfigBadValue` message, so, no, it'll be there if someone can see the traceback.

> The use of `try`/`else` is an interesting quirk

Yes...Peter tends to use it as though the `else` were required...which leaves one wondering when/how we can escape that block and get back to the previous level of indentation...so I tend to eschew the `try`/`else` construction except when it's actually needed (i.e., when the code actually recovers from an exception, and you don't want the recovery to execute the same path as the normal exit from the `try`).

> it could have just raised `ServerConfigBadValue`

Yes, but I was trying to combine the two failure cases into a single one....